hazelcast/src/main/java/com/hazelcast/partition/strategy/StrategyUtil.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.partition.strategy;
+
+public final class StrategyUtil {
+    private StrategyUtil() { }
+
+    public static Object constructKey(Object[] keyComponents) {
+        return keyComponents.length == 1 ? keyComponents[0] : keyComponents;
```

Comment:
this logic is fine and specific for `AttributePartitionStrategy`, but is not general. Method name and/or javadoc should indicate that.